pkg/reconciler/instances/ory/action.go

```diff
@@ -163,6 +191,27 @@ func (a *postAction) patchSecret(ctx context.Context, client kubernetes.Interfac
 	return err
 }
 
+func (a *preDeleteAction) dbSecretExists(ctx context.Context, client kubernetes.Interface, name types.NamespacedName) (bool, error) {
+	_, err := client.CoreV1().Secrets(name.Namespace).Get(ctx, name.Name, metav1.GetOptions{})
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, errors.Wrap(err, "Could not get DB secret")
+	}
+	return true, nil
+}
+
+func (a *preDeleteAction) deleteSecret(ctx context.Context, client kubernetes.Interface, name types.NamespacedName, logger *zap.SugaredLogger) error {
+	err := client.CoreV1().Secrets(name.Namespace).Delete(ctx, name.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to delete the secret")
```

Comment:
I don't know the API, but perhaps it's worth to check if `if kerrors.IsNotFound(err)` and if so, return true here.
It is one of the common network programming errors: You assume that between the check if the secret exists (time == t[0]) until the moment you actually request a delete (time == t[1] == t[0]+delta), nothing changes.
What if your delta is 10 seconds? Easily possible in network scenarios. Ordinary user can delete the secret by hand within that time.

Reply:
It's an edge case. But a valid one. Will do.